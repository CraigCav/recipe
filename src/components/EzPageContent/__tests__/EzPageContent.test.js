import React from 'react';
import {EzPageContent, EzContentGroup} from '../';

describe('EzPageContent', () => {
  it('should render with default styles', () => {
    const actual = create(
      <EzPageContent>
        <p>Lorem ipsum dolor</p>
        <p>Lorem ipsum dolor</p>
      </EzPageContent>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should render EzContentGroup with default styles', () => {
    const actual = create(
      <EzPageContent>
        <EzContentGroup>
          <p>Lorem ipsum dolor</p>
          <p>Lorem ipsum dolor</p>
        </EzContentGroup>
      </EzPageContent>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should render EzContentGroup content with horizontal styles', () => {
    const actual = create(
      <EzPageContent>
        <EzContentGroup horizontal>
          <p>Lorem ipsum dolor</p>
          <p>Lorem ipsum dolor</p>
        </EzContentGroup>
      </EzPageContent>
    );
    expect(actual).toMatchSnapshot();
  });

  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(
      <EzPageContent>
        <EzContentGroup>
          <p>Lorem ipsum dolor</p>
          <p>Lorem ipsum dolor</p>
        </EzContentGroup>
      </EzPageContent>
    );
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
