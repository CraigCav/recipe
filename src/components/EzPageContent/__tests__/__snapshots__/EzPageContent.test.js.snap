// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzPageContent should render EzContentGroup content with horizontal styles 1`] = `
.recipe-2 {
  background: #f4f7f8;
  padding: 1.5rem;
}

.recipe-2 > *:not(:first-child) {
  margin-top: 3rem;
}

.recipe-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.recipe-0 > * {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.recipe-0 > *:not(:first-child) {
  margin-left: 1.5rem;
}

<div
  className="recipe-2 recipe-3"
>
  <div
    className="recipe-0 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
</div>
`;

exports[`EzPageContent should render EzContentGroup with default styles 1`] = `
.recipe-2 {
  background: #f4f7f8;
  padding: 1.5rem;
}

.recipe-2 > *:not(:first-child) {
  margin-top: 3rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.5rem;
}

<div
  className="recipe-2 recipe-3"
>
  <div
    className="recipe-0 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
</div>
`;

exports[`EzPageContent should render with default styles 1`] = `
.recipe-0 {
  background: #f4f7f8;
  padding: 1.5rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 3rem;
}

<div
  className="recipe-0 recipe-1"
>
  <p>
    Lorem ipsum dolor
  </p>
  <p>
    Lorem ipsum dolor
  </p>
</div>
`;
